src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,45 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        File file = new File(fromFileName);
+        int totalSupply = 0;
+        int totalBuy = 0;
+
+        try {
+            List<String> rows = Files.readAllLines(file.toPath());
+            for (String row : rows) {
+                String[] parts = row.split(",");
+                String operation = parts[0];
+                int value = Integer.parseInt(parts[1]);
+
+                if (operation.equals("supply")) {
+                    totalSupply += value;
+                } else if (operation.equals("buy")) {
+                    totalBuy += value;
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Unable to read provided file: " + fromFileName, e);
+        }
+
+        int totalResult = totalSupply - totalBuy;
+
+        File finalFile = new File(toFileName);
```

Comment:
Before writing to the file, consider checking if the file is writable or if there are any permission issues. This can prevent runtime exceptions during file writing operations.